cabal-install/src/Distribution/Client/Init/Types.hs

```diff
@@ -422,6 +424,19 @@ type IsLiterate = Bool
 --
 type IsSimple = Bool
 
+-- | Defines whether or not if a prompt will have a default value,
+--   is optional, or is mandatory.
+data DefaultPrompt t
+  = DefaultPrompt t
+  | OptionalPrompt
+  | MandatoryPrompt
```

Comment:
There is a convention that the common part of constructor name is a prefix. However, the suffix here reads very naturally, so perhaps this version is fine. Also, people nowadays use IDEs, so they don't need a prefix to know what the type is.